### PR TITLE
DEV-2595: Fix redirect link and update error handling

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a

#### What's this PR do?
Fixes bug in which Org Admin user doesn't redirected to new page on page creation

#### Why are we doing this? How does it help us?
 Fixes the Create new page flow

#### How should this be manually tested? Please include detailed step-by-step instructions.
- Create a new account
- Verify email and fill in profile inputs
- Create new page
- Make sure the flow works as expected (page gets created and user is redirected to edit page)
- User cannot create a second page and error pops up for the user to know what happened

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
No

#### Have automated unit tests been added? If not, why?
No, the create page modal/flow will be updated in the next ticket (DEV-2299 Update and Enhance "New page" modal)

#### How should this change be communicated to end users?
n/a

#### Are there any smells or added technical debt to note?
Create Page modal/flow isn't really well written. Will be addressed in DEV-2299

#### Has this been documented? If so, where?
no

#### What are the relevant tickets? Add a link to any relevant ones.
https://news-revenue-hub.atlassian.net/browse/DEV-2595

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
no

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
https://news-revenue-hub.atlassian.net/browse/DEV-2299

### CODE REVIEW

File: spa/src/components/content/pages/AddPageModal.js [hunk 1]
```diff
@@ -54,7 +54,11 @@ function AddPageModal({ isOpen, closeModal, pagesByRevenueProgram }) {
     (e) => {
       setLoading(false);
       if (e.response?.data) {
-        setErrors(e.response.data);
+        if (revenuePrograms?.length === 1) {
```
Comment (nrh-cklimas) [inline spa/src/components/content/pages/AddPageModal.js:57]: Is this the bug fix? e.g. before it was failing silently?

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]
I get that we are planning to revamp the modals soon, but I’d prefer to have some test coverage on this if at all possible. Could you add some really basic unit tests for the bug this story is about? They don't need to be elaborate if you'll be revisiting the modal soon.

File: spa/src/components/content/pages/AddPageModal.js (hunk 1 above)
Comment (GuiMend) [reply spa/src/components/content/pages/AddPageModal.js:57]: Yes, before it wouldn't show the error `you have reached the page limit of 1` (or something like this)

Comment (GuiMend) [reply spa/src/components/content/pages/AddPageModal.js:57]: added tests for this

**Review by nrh-cklimas — APPROVED** [review]
Looks good to me now!

**Review by x110dc — APPROVED** [review]

**Pull request merged**